android-auto-app/src/main/AndroidManifest.xml

```diff
@@ -38,6 +38,14 @@
             android:value="1"
             tools:ignore="MetadataTagInsideApplicationTag" />
 
+        <meta-data
+            android:name="com.mapbox.maps.ThreadChecker"
+            android:value="false" />
+
+        <meta-data
+            android:name="com.mapbox.maps.BackgroundAnimatorThread"
+            android:value="true" />
```

Comment:
Although background thread is used for animations - gl-native calls should still happen from the main thread. Thus this could be omitted I guess.

Another point - IMO we don't need to actually switch to background thread for animations on _all_ the devices for AA. On most devices (not Xiaomi) it works OK as it is now. It makes sense imo to hide switching animations to background thread under some API and delegate setting this flag to end users. Switching to background thread is not something safe and we should retain current behaviour (using main thread to start animations) as the default one to reduce errors.
cc @yunikkk as we were discussing that offline

Reply:
Notice that this is only enabled for the `android-auto-app`. The idea is to have zero impact on the current main thread solution. The developers who are trying to release android-auto could enable the flag, to make it work.

That said, there are still a lot of issues in this PR with thread context switching. It locks, it janks, and it crashes unexpectedly. Unfortunately, the degraded experience is still better than what is happening now - it does not work at all.

Do you @yunikkk have a device that can reproduce this error? It is not a small amount of work.